gc.rb

```diff
@@ -250,71 +250,58 @@ def self.stat_heap heap_name = nil, hash_or_key = nil
+  #   GC.config(rgengc_allow_full_mark: false)
+  #   # => {rgengc_allow_full_mark: false}
+  #   GC.config
+  #   # => {rgengc_allow_full_mark: false, implementation: "default"}
+  #   GC.config(foo: 'bar')
+  #   # => {rgengc_allow_full_mark: false}
+  #   GC.config
+  #   # => {rgengc_allow_full_mark: false, implementation: "default"}
   #
-  # This method is only expected to work on CRuby.
-  #
-  # === \GC Implementation independent values
-  #
-  # The <code>GC.config</code> hash can also contain keys that are global and
-  # read-only. These keys are not specific to any one \GC library implementation
-  # and attempting to write to them will raise +ArgumentError+.
+  # <b>All-Implementations Configuration</b>
   #
-  # There is currently only one global, read-only key:
+  # The single read-only entry for all implementations is:
   #
-  # [implementation]
-  #    Returns a +String+ containing the name of the currently loaded \GC library,
-  #    if one has been loaded using +RUBY_GC_LIBRARY+, and "default" in all other
-  #    cases
+  # - +implementation+:
+  #   the string name of the implementation;
+  #   for the Ruby default implementation, <tt>'default'</tt>.
   #
-  # === \GC Implementation specific values
+  # <b>Implementation-Specific Configuration</b>
   #
-  # \GC libraries are expected to document their own configuration. Valid keys
-  # for Ruby's default \GC implementation are:
+  # A \GC implementation maintains its own implementation-specific configuration.
   #
-  # [rgengc_allow_full_mark]
-  #   Controls whether the \GC is allowed to run a full mark (young & old objects).
+  # For Ruby's default implementation the single entry is:
   #
-  #   When +true+, \GC interleaves major and minor collections. This is the default. \GC
-  #   will function as intended.
+  # - +rgengc_allow_full_mark+:
+  #   Controls whether the \GC is allowed to run a full mark (young & old objects):
   #
-  #   When +false+, the \GC will never trigger a full marking cycle unless
-  #   explicitly requested by user code. Instead, only a minor mark will run—
-  #   only young objects will be marked. When the heap space is exhausted, new
-  #   pages will be allocated immediately instead of running a full mark.
+  #   - +true+ (default): \GC interleaves major and minor collections.
+  #   - +false+: \GC does not initiate a full marking cycle unless
+  #     explicitly directed by user code;
+  #     see GC.start.
   #
-  #   A flag will be set to notify that a full mark has been
-  #   requested. This flag is accessible using
-  #   <code>GC.latest_gc_info(:need_major_by)</code>
-  #
-  #   The user can trigger a major collection at any time using
-  #   <code>GC.start(full_mark: true)</code>
-  #
-  #   When +false+, Young to Old object promotion is disabled. For performance
-  #   reasons, it is recommended to warm up an application using +Process.warmup+
+  #   Setting this parameter to +false+ disables young-to-old promotion .
```

Comment:
Nit: space before period

```suggestion
  #   Setting this parameter to +false+ disables young-to-old promotion.
```

Reply:
Fixed.